components/net/connector.rs

```diff
@@ -50,35 +50,20 @@ impl NetworkConnector for HttpsConnector {
 
 pub type Connector = HttpsConnector;
 
-pub fn create_ssl_client(ca_file: &PathBuf) -> OpensslClient {
-    let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
-    {
-        let context = ssl_connector_builder.builder_mut();
-        context.set_ca_file(ca_file).expect("could not set CA file");
-        context.set_cipher_list(DEFAULT_CIPHERS).expect("could not set ciphers");
-        context.set_options(SSL_OP_NO_SSLV2 | SSL_OP_NO_SSLV3 | SSL_OP_NO_COMPRESSION);
-    }
-    let ssl_connector = ssl_connector_builder.build();
-    OpensslClient::from(ssl_connector)
+pub fn create_ssl_client(ca_file: &PathBuf) -> TlsClient {
+    let mut ca = {
+        let f = fs::File::open(ca_file).unwrap();
+        let rd = io::BufReader::new(f);
+        rd
+    };
+    let mut tls = rustls::ClientConfig::new();
+    tls.root_store.add_pem_file(&mut ca).unwrap();
+    let tls = Arc::new(tls);
+    let client = TlsClient { cfg: tls };
```

Comment:
`TlsClient { cfg: Arc::new(tls) }`
Rust automatically returns the last expression of a function: https://rustbyexample.com/fn.html